### PR TITLE
Fixed tests 

### PR DESCRIPTION
**Pull request by OmarSemou**
**TLDR:**
The primary issue stemmed from the Id field for authors being treated as a string, leading to incorrect sorting and ID incrementation logic. The changes make tests now pass and behave as expected by treating Id as a numeric value. 

**Long version:**

1. Updated GetLatestIdAuthor Method:
     - Fixed the logic to ensure numeric sorting of author Id values by:
        - Fetching all author records into memory using ToListAsync.
        - Parsing Id values as integers for proper numeric sorting.
     - This resolves issues with lexicographical sorting of string IDs (for example., "121" being treated as smaller than "16").

2. Fixed CreateAuthor Method:

     - Corrected ID incrementation by:
        - Parsing the latest Id as an integer.
        - Performing numeric addition and converting the result back to a string.
     - Ensured new authors are assigned the correct numeric increment of the highest existing Id.

### CODE REVIEW

**Review by Asger124 — APPROVED** [review]
Nice job in fixing the tests, and also altering the methods to properply handle the ID we now get from identity users 

I am wondering, you have changed some stuff in the workflow as well - looks like it has to do with GithUb login. Did this affect the tests too, or is this just an addition you have made in this PR ?

**Comment by OmarSemou** [comment]
Yeah I altered the workflow because it was necessary to provide secrets in order for oAuth to work, our integration tests fail otherwise.
